install/install.go

```diff
@@ -37,11 +37,33 @@ type Options struct {
 	Force          bool
 	Save           bool
 	Animation      bool
+	Env            EnvOption
+}
+
+type EnvOption struct {
+	Auto        bool
+	Environment string
 }
 
 var animation bool
 var progress int
 var cursorcount int
+var validEnvironments = []string{"k3s", "microK8s", "minikube", "gke", "bottlerocket", "eks", "docker", "oke", "generic"}
+
+func (env *EnvOption) CheckAndSetValidEnvironmentOption(envOption string) error {
+	for _, v := range validEnvironments {
+		if v == envOption {
+			env.Environment = envOption
+			env.Auto = false
+			return nil
+		}
+	}
+	env.Auto = true
+	if envOption == "" {
+		return nil
+	}
```

Comment:
Shouldn't this check come before we check for valid  Environments?
```suggestion
	if envOption == "" {
	        env.Auto = true
		return nil
	}
	for _, v := range validEnvironments {
		if v == envOption {
			env.Environment = envOption
			env.Auto = false
			return nil
		}
	}
```

Reply:
yeah , i will fix that